fe/fe-core/src/main/java/com/starrocks/common/ConfigBase.java

```diff
@@ -75,29 +80,38 @@ public class ConfigBase {
         String[] aliases() default {};
     }
 
-    protected Properties props;
+    protected Properties props = new Properties();
+    private static String mutableConfigPath;
+    private static boolean isPersisted = false;
     protected static Field[] configFields;
     protected static Map<String, Field> allMutableConfigs = new HashMap<>();
 
     public void init(String propFile) throws Exception {
         configFields = this.getClass().getFields();
-        initAllMutableConfigs();
-        props = new Properties();
-        FileReader reader = null;
-        try {
-            reader = new FileReader(propFile);
+        try (FileReader reader = new FileReader(propFile)) {
             props.load(reader);
-        } catch (Exception e) {
-            throw e;
-        } finally {
-            if (reader != null) {
-                reader.close();
-            }
         }
         replacedByEnv();
+    }
+
+    public void initMutable(String propFile) throws Exception {
+
+        Path path = Path.of(propFile);
+        if (!Files.exists(path)) {
+            Files.createFile(path);
+        }
+        mutableConfigPath = propFile;
+        initAllMutableConfigs();
+        try (FileReader reader = new FileReader(mutableConfigPath)) {
+            props.load(reader);
+        }
         setFields();
+        if (Files.isWritable(path)) {
```

Comment:
Have you tested this feature in FE HA mode, does the follower can get the correct configuration and persists the same as the leader?

Especially when the leader is writable but the follower is not writable or the vice versa

Reply:
To ensure data consistency, it is best that all nodes can write to the configuration before enabling this feature.

The complexity here lies in how to synchronize information across all nodes; should we use metadata? or are there other methods? Do you have any better suggestions?